src/services/address.service.ts

```diff
@@ -35,36 +36,211 @@ class AddressService {
         });
     }
 
+
+    /**
+     * Calculate the distance between two locations.
+     * @param addressRequest - The request object containing location1 and location2.
+     * 
+     * The body request should contain:
+     * - location1: The first location object. (Can mix and match any of the following properties)
+     * > - zipcode: The zipcode of the location.
+     * > - number: The number of the location.
+     * > - street: The street of the location.
+     * > - street2: The second street of the location.
+     * > - city: The city of the location.
+     * > - state: The state of the location.
+     * > - country: The country of the location.
+     * - location2: The second location object.
+     * > - Same properties as location1
+     * - unit: The unit of measurement for the distance (KM or Mi).
+     * 
+     * @returns The distance between the two locations in the specified unit (KM or Mi).
+     * 
+     */
     public async distance(addressRequest?: any): Promise<any> {
-        // Complete this
+
+        if (addressRequest.body) {
+            try {
+                let location1 = addressRequest.body.location1;
+                let location2 = addressRequest.body.location2;
+
+                // Check if both locations are provided
+                if (!location1 || !location2) {
+                    const errMessage = "Please provide both locations.";
+                    loggerService.error({ path: "/address/distance", message: errMessage }).flush();
+                    return { error: errMessage };
+                }
+
+                // Check if the locations are valid
+                const requiredProperties = [
+                    "zipcode", "number", "street", "street2", 
+                    "city", "state", "country"
+                ];
+
+                const hasRequiredProperties = (location: any) => 
+                    requiredProperties.some(prop => location.hasOwnProperty(prop));
+
+                if (
+                    typeof location1 !== "object" || 
+                    typeof location2 !== "object" || 
+                    !hasRequiredProperties(location1) || 
+                    !hasRequiredProperties(location2)
+                ) {
+                    const errMessage = "Invalid locations. Please ensure both locations have at least one required property. Required properties: " + requiredProperties.join(", ");
+                    loggerService.error({ path: "/address/distance", message: errMessage }).flush();
```

Comment:
Is this an error for this app or the client that made the request? I'd say sending the message back to the client is good, but maybe log this as a warning within this app instead.